File: problem_010.lua
#!/usr/bin/env lua

function isPrime (number)
   limit = math.floor(math.sqrt(number))
   for index = 3, limit , 2 do
      if (number % index == 0) then
         return false
      end
   end
   return true
end

print("Project Euler - Problem 10:\n")
print("Find the sum of all the primes below two million.\n\n")

sum = 2 --Starting off with first prime accounted for
for number=3, 2000000, 2 do
   if (isPrime(number)) then
      sum = sum + number
   end
end

print("Sum of primes: "..sum.."\n")

File: problem_010.py
#!/usr/bin/env python

print("Project Euler - Problem 10:")
print("Find the sum of all the primes below two million.\n")

limit = 2000000
primes = [0, 0] + [1] * (limit-2)

for i in range(len(primes)):
    if primes[i] == 1:
        primes[i] = i
        n = 2
        while n * i < limit:
            primes[n * i] = 0
            n += 1

print(sum(primes))
